src/editables/__init__.py

```diff
@@ -35,29 +44,29 @@ def map(self, name, target):
         else:
             raise EditableException(f"{target} is not a valid Python package or module")
 
-    def add_to_path(self, dirname):
+    def add_to_path(self, dirname: "str | os.PathLike[str]") -> None:
         self.path_entries.append(self.make_absolute(dirname))
 
-    def files(self):
+    def files(self) -> "Iterator[tuple[str, str]]":
```

Comment:
Why have you quoted the return type?

Reply:
Because built-in generics are not subscriptable in Python < 3.9 and we cannot use `from __future__ import annotations` in 3.6.

Reply:
The alternative of course would be to use the now deprecated types from `typing`.

Reply:
OK, let's use those then. I find quoted type annotations ugly.